cpp/src/io/functions.cpp

```diff
@@ -338,6 +338,34 @@ parsed_orc_statistics read_parsed_orc_statistics(source_info const& src_info)
   return result;
 }
 
+orc_column_metadata make_orc_column_metadata(host_span<orc::SchemaType const> orc_schema,
```

Comment:
Can this return type be const? Should anyone alter the metadata returned?

Reply:
Thought about this and realized that we should encapsulate members of the user-facing types here. Same effect as const, and more! :)